### PR TITLE
Added Python 3.10, removed 3.7 support, streamlined tests

### LINKED ISSUE #869: Ensure clean python 3.10.0 build path
I'm having issues installing graspologic on 3.10.0 right now (see fun screenshot). This may or may not be local to my computer.

In any case, we should ensure we have a viable build path for graspologic on 3.10.0. See @daxpryce's advice here:

> We can also enable 3.10 in our build matrix on a branch and see what our github runners are saying across all 3 OSes so we can collect all of our problems at once :)

Screen Shot 2021-10-29 at 3 49 21 PM

### LINKED ISSUE #922: [BUG] Graspologic does not work on Python 3.10
## Expected Behavior
We should be able to run our entire test suite using Python 3.10

## Actual Behavior
```python
graspologic/embed/n2v.py:11: in <module>
    from gensim.models import Word2Vec
/opt/hostedtoolcache/Python/3.10.2/x64/lib/python3.10/site-packages/gensim/__init__.py:11: in <module>
    from gensim import parsing, corpora, matutils, interfaces, models, similarities, utils  # noqa:F401
/opt/hostedtoolcache/Python/3.10.2/x64/lib/python3.10/site-packages/gensim/corpora/__init__.py:6: in <module>
    from .indexedcorpus import IndexedCorpus  # noqa:F401 must appear before the other classes
/opt/hostedtoolcache/Python/3.10.2/x64/lib/python3.10/site-packages/gensim/corpora/indexedcorpus.py:14: in <module>
    from gensim import interfaces, utils
/opt/hostedtoolcache/Python/3.10.2/x64/lib/python3.10/site-packages/gensim/interfaces.py:19: in <module>
    from gensim import utils, matutils
/opt/hostedtoolcache/Python/3.10.2/x64/lib/python3.10/site-packages/gensim/matutils.py:10[24](https://github.com/microsoft/graspologic/runs/5147181977?check_suite_focus=true#step:5:24): in <module>
    from gensim._matutils import logsumexp, mean_absolute_difference, dirichlet_expectation
gensim/_matutils.pyx:1: in init gensim._matutils
    ???
E   ValueError: numpy.ndarray size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject
```
From: https://github.com/microsoft/graspologic/runs/5147180666?check_suite_focus=true#step:5:1

### LINKED ISSUE #998: [BUG] giant gensim error when trying to pip install on 3.10.6
`pip install graspologic`

Results in a giant error having to do with the `gensim` requirement.
I'm using a fresh-ish ubuntu environment with python 3.10.6

I'd copy the whole error, but it's like 10 years long and all having to do with weird unicode stuff.
Is gensim a real requirement?

gensim_error

### PR DESCRIPTION
**Pull request by bdpedigo**
- [x] Does this PR have a descriptive title that could go in our release notes?
- [ ] Does this PR add any new dependencies?
- [x] Does this PR modify any existing APIs?
   - [x] Is the change to the API backwards compatible?
- [ ] Have you built the documentation (reference and/or tutorial) and verified the generated documentation is appropriate?

#### Reference Issues/PRs
Closes #922 
Closes #869
Closes #998 (I think) @loftusa you are welcome to test

#### What does this implement/fix? Briefly explain your changes.
- Adds Python 3.10 support since we're really behind on that
- Removes Python 3.7 support since that is considered end of life and modern numpy is no longer supporting it. Would like to be able to use new features of scipy, for instance, in https://github.com/microsoft/graspologic/pull/992 
- Bumped the major version because of the above
- I couldn't stand super long tests any longer so I stripped down many of the super slow simulation tests
   - By way of explanation, I dont think the tests should be designed to make sure everything is statistically valid etc. so I removed many that were running long simulations to ensure this.

#### Any other comments?
- Will try 3.11 in a PR after this if this one works
- 3.8 is still used for most things like building docs, should we bump that as well?

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [metadata]
 name = graspologic
-version = 2.0.1
+version = 3.0.0
```
Comment (bdpedigo) [inline setup.cfg:3]: believe not having python 3.7 support warrants a major version bump?

Comment (daxpryce) [reply setup.cfg:3]: agreed, it drives me nuts that they don't do this for numpy/scipy when it clearly is a major change that will break your users

**Review by daxpryce — APPROVED** [review]
Looks good to me!

**Comment by bdpedigo** [comment]
thanks @daxpryce - it looks like the only thing left is the 3.7 tests, which no longer get run anymore. 

does this require god powers to merge?

**Comment by bdpedigo** [comment]
🙏

**Comment by bdpedigo** [comment]
idk if we care but looks like commits were not squashed on that merge

**Pull request merged**